test/catch2/live_restore/utils_live_restore.cpp

```diff
@@ -0,0 +1,86 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "utils_live_restore.h"
+#include <fstream>
+namespace utils {
+
+/*
+ * Create a string representation of an extent list, for example (1-10),(15-30) represents an extent
+ * list with holes at ranges 1 to 10 and 15 to 30 (inclusive).
+ */
+std::string
+extent_list_str(WT_LIVE_RESTORE_FILE_HANDLE *lr_fh)
+{
+    WT_LIVE_RESTORE_HOLE_NODE *ext = lr_fh->destination.hole_list_head;
+
+    std::string str = "";
+
+    if (ext == NULL) {
+        // NULL is an empty list
+        return "";
+    }
+
+    while (ext != NULL) {
+        str += "(" + std::to_string(ext->off) + "-" + std::to_string(WT_EXTENT_END(ext)) + "), ";
+        ext = ext->next;
+    }
+
+    // Remove the trailing ", "
+    if (str.size() > 0) {
+        str.erase(str.size() - 2);
+    }
+    return str;
+}
+
+/* Open the live restore file handle for a file. This file path is identical to the backing file in
+ * the destination folder. */
+void
+open_lr_fh(const live_restore_test_env &env, const std::string &dest_file,
+  WT_LIVE_RESTORE_FILE_HANDLE **lr_fhp)
+{
+    WT_LIVE_RESTORE_FS *lr_fs = env._lr_fs;
+    WT_SESSION *wt_session = reinterpret_cast<WT_SESSION *>(env._session);
+    // Make sure we're always opening the file in the destination directory.
+    REQUIRE(strncmp(dest_file.c_str(), env._DB_DEST.c_str(), env._DB_DEST.size()) == 0);
+    testutil_check(lr_fs->iface.fs_open_file(reinterpret_cast<WT_FILE_SYSTEM *>(lr_fs), wt_session,
+      dest_file.c_str(), WT_FS_OPEN_FILE_TYPE_REGULAR, 0,
+      reinterpret_cast<WT_FILE_HANDLE **>(lr_fhp)));
+}
+
+/* Verify that all extents in an extent list are in order and don't overlap. */
+bool
+extent_list_in_order(WT_LIVE_RESTORE_FILE_HANDLE *lr_fh)
+{
+    WT_LIVE_RESTORE_HOLE_NODE *prev_node, *node;
+    prev_node = NULL;
+    node = lr_fh->destination.hole_list_head;
+    while (node != NULL) {
+        if (prev_node != NULL) {
+            if (prev_node->off >= node->off || WT_EXTENT_END(prev_node) >= node->off) {
+                return false;
+            }
+        }
+        prev_node = node;
+        node = node->next;
+    }
```

Comment:
I prefer the below code for the while loop but I am fine with yours, you can ignore this suggestion if you don't like it.
```
while (node != NULL && node->next != NULL) {
    if (WT_EXTENT_END(node) >= node->next->off) {
        return false;
    }
    node = node -> next;
}
```

Reply:
Looks good! I've taken it wholesale in [e043f5c](https://github.com/wiredtiger/wiredtiger/pull/11332/commits/e043f5c995436248669fa08c5bdeb4bb791a0973)